ines-test-base.py

```diff
@@ -0,0 +1,81 @@
+import pytest
+from sklearn.base import BaseEstimator
+from sklearn.naive_bayes import GaussianNB
+from sklearn.preprocessing import StandardScaler
+from sklearn.linear_model import LogisticRegression
+from sklearn.svm import SVC
+from sklearn.tree import DecisionTreeClassifier
+from sklearn.cluster import KMeans
+from packaging.version import parse as parse_version
+import sklearn
+
+class CustomEstimator(BaseEstimator):
+    _doc_link = "https://custom_link/{major}.{minor}/docs/{estimator_name}.html"
+    _doc_link_module = "custom_module"
+
+    def _get_url_link(self):
+        major = 2
+        minor = 19
+        estimator_name = self.__class__.__name__
+        full_url = self._doc_link.format(**locals())
+        return full_url
+    def __init__(self, parameter1=None, parameter2=None):
+        # Initialize any required parameters
+        self.parameter1 = parameter1
+        self.parameter2 = parameter2
+
+    def fit(self, X, y=None):
+        # Implement the fitting logic of your estimator
+        # ...
+        return self
+
+    def predict(self, X):
+        # Implement the prediction logic of your estimator
+        # ...
+        return X
+class CustomEstimator_wrong(BaseEstimator):
+    def __init__(self, parameter1=None, parameter2=None):
+        # Initialize any required parameters
+        self.parameter1 = parameter1
+        self.parameter2 = parameter2
+
+    def fit(self, X, y=None):
+        # Implement the fitting logic of your estimator
+        # ...
+        return self
+
+    def predict(self, X):
+        # Implement the prediction logic of your estimator
+        # ...
+        return X
+
+def _local_get_estimator_doc_url(estimator):
+    """Generating a link to the API documentation for a given estimator and scikit-learn version."""
```

Comment:
This seems like a copy of the code. We shouldn't do this. We need to test at a slightly higher level, eg call '_repr_html' on an estimator, and check that we have a link to "http://scikit-learn.org" in it. This will be a less precise test indeed